### PR TITLE
Make return type of ProtoAdapter.redact() non-nullable

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ abstract class ProtoAdapter<E>(
   internal var repeatedAdapter: ProtoAdapter<List<E>>? = null
 
   /** Returns the redacted form of `value`. */
-  open fun redact(value: E): E? = null
+  open fun redact(value: E): E = throw UnsupportedOperationException()
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Do you just want to make this `abstract`?

Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Yep, good point

Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: That breaks binary compat though.. I'm gonna merge as is, can revisit later

Comment (JakeWharton) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: It's a major version bump though. Generated code is only guaranteed to work with the same compiler version.

Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Well yeah, guess that's fine.

**Pull request merged**